pkg/collector/corechecks/servicediscovery/impl_linux.go

```diff
@@ -87,19 +85,24 @@ func (li *linuxImpl) DiscoverServices() (*discoveredServices, error) {
 		}
 	}
 
-	ports, err := li.portPoller.OpenPorts()
+	sysProbe, err := li.getSysProbeClient()
 	if err != nil {
 		return nil, errWithCode{
 			err:  err,
-			code: errorCodePortPoller,
-			svc:  nil,
+			code: errorCodeSystemProbeConn,
 		}
 	}
-	portsByPID := map[int]portlist.List{}
-	for _, p := range ports {
-		portsByPID[p.Pid] = append(portsByPID[p.Pid], p)
+
+	response, err := sysProbe.GetDiscoveryListeners()
+	if err != nil {
+		return nil, errWithCode{
+			err:  err,
+			code: errorCodeSystemProbeOpenPorts,
+		}
 	}
 
+	log.Debugf("listeners: %v", response.Listeners)
```

Comment:
How often is DiscoverServices being called?
if once in 1m, then ignore this comment
if more frequent, then we should wrap this call with a check to the log level, as we've discovered in the past that called `log.Debug` when the log level is info or above, still allocates redundant memory.